frappe/database/database.py

```diff
@@ -79,7 +78,7 @@ def __init__(
 		self.setup_type_map()
 		self.host = host or frappe.conf.db_host
 		self.port = port or frappe.conf.db_port
-		self.user = user or frappe.conf.db_name
+		self.user = user or frappe.conf.db_user or frappe.conf.db_name
```

Comment:
This fallback shouldn't be required?

Reply:
There was one case where it wasn't getting set here